pkg/store/bucket.go

```diff
@@ -2033,7 +2042,8 @@ func (s *BucketStore) LabelValues(ctx context.Context, req *storepb.LabelValuesR
 
 		// If we have series matchers and the Label is not an external one, add <labelName> != "" matcher
 		// to only select series that have given label name.
-		if len(reqSeriesMatchersNoExtLabels) > 0 && !b.extLset.Has(req.Label) {
+		// We don't need such matcher if matchers already contain __name__ matcher.
```

Comment:
Trying to understand this comment. Why it is specific to the metric name label?
It is just a tradeoff of fetching more postings or series.

Reply:
From my experience, if `__name__` is specified, it means, user knows, that the metric contains requested label. The method will select all series with such `__name__` and they 99% will have the requested label. 
As for random queries, where there can be results with the `__name__` but without specified labels, normally, they should be rare. Users still can make queries like `label_values({}, pod)`. They will work, but will fetch alot of data.
So the point is, if the user knows what they need and specifies `__name__`, we don't need to save them from fetching some extra series. But we do save them from fetching all references to all kube_state_metrics from object storage for example.
Other popular labels may be `service`, `application`, `job`, `instance`. If `__name__` is specified, it is guaranteed it'll be less data, but all the label values.

Reply:
Not against this. Just hope we have a better way to cover more labels because we have those information from posting cardinality and series size.